evalml/pipelines/components/estimators/regressors/timeseries_baseline_regressor.py

```diff
@@ -0,0 +1,74 @@
+import numpy as np
+import pandas as pd
+
+from evalml.model_family import ModelFamily
+from evalml.pipelines.components.estimators import Estimator
+from evalml.problem_types import ProblemTypes
+from evalml.utils.gen_utils import (
+    _convert_to_woodwork_structure,
+    _convert_woodwork_types_wrapper
+)
+
+
+class TimeSeriesBaselineRegressor(Estimator):
+    """Time series regressor that predicts using the naive forecasting approach.
+
+    This is useful as a simple baseline regressor for time series problems
+    """
+    name = "Time Series Baseline Regressor"
+    hyperparameter_ranges = {}
+    model_family = ModelFamily.BASELINE
+    supported_problem_types = [ProblemTypes.TIME_SERIES_REGRESSION]
+
+    def __init__(self, random_state=0, **kwargs):
+        """Baseline time series regressor that predicts using the naive forecasting approach.
+
+        Arguments:
+            random_state (int, np.random.RandomState): seed for the random number generator
+
+        """
+
+        self._prediction_value = None
+        self._num_features = None
+        self.y_hat = None
+
+        parameters = {}
+        parameters.update(kwargs)
+        super().__init__(parameters=parameters,
+                         component_obj=None,
+                         random_state=random_state)
+
+    def fit(self, X, y=None):
+        if X is None:
+            X = pd.DataFrame()
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+
+        self._num_features = X.shape[1]
+        return self
+
+    def predict(self, X, y=None):
+        if X is None:
+            X = pd.DataFrame()
+        if y is None:
+            raise ValueError("Cannot predict Time Series Baseline Regressor if y is None")
+        X = _convert_to_woodwork_structure(X)
+        X = _convert_woodwork_types_wrapper(X.to_dataframe())
+        y = _convert_to_woodwork_structure(y)
+        y = _convert_woodwork_types_wrapper(y.to_series())
+
+        first = y.iloc[0]
```

Comment:
I don't think we should do any shifting here and `predict` should basically be a no-op that return the `y` that is input. 

The `score` method in `TimeSeriesRegressionPipeline` will shift the target by `-gap` which will ensure the target is `gap` time periods ahead of the prediction. Right now, with this `shift` they would be offset by `gap + 1` which I don't think is intended. Let's say the user has dates October 1-8 and the gap is 3. I think the desired pairs of prediction and target dates should be:

| Prediction Date      | Target Date |
| ----------- | ----------- |
| 2020-10-01      | 2020-10-04       |
| 2020-10-02   | 2020-10-05        |
| 2020-10-03 | 2020-10-06|
| 2020-10-04 | 2020-10-07|
| 2020-10-05 | 2020-10-08|

We need to be mindful of the `gap=0` case, cause we should shift by `1` in this `predict` method to avoid perfect overlap in the pipeline `score` method.

Reply:
@freddyaboulton  got it! will make the changes.